sdk/core/azure_core_tracing_opentelemetry/README.md

```diff
@@ -0,0 +1,92 @@
+# Azure Core OpenTelemetry Tracing
+
+This crate provides OpenTelemetry distributed tracing support for the Azure SDK for Rust. It enables automatic span creation, context propagation, and telemetry collection for Azure service operations.
+
+## Features
+
+- **Automatic span creation** for Azure service operations
+- **Context propagation** across service boundaries
+- **Azure-specific conventions** for span attributes
+- **Error tracking** with detailed error information
+- **Performance metrics** including operation duration
+- **Configurable telemetry** with multiple output options
+
+## Usage
+
+### Basic Setup
+
+```rust
+use azure_core_tracing_opentelemetry::{init_tracing, TelemetryConfig, AzureTelemetryProvider};
+
+// Initialize with default configuration
+init_tracing()?;
+
+// Or with custom configuration
+let config = TelemetryConfig::new("my-azure-app")
+    .with_service_version("1.0.0")
+    .with_log_level("debug");
+
+let provider = AzureTelemetryProvider::new(config)?;
+provider.install_global()?;
+```
+
+### Creating Spans
+
+```rust
+use azure_core_tracing_opentelemetry::{create_span, SpanBuilder, SpanKind, Attribute, AttributeValue};
+
+// Simple span creation
+let span = create_span("storage_operation", vec![
+    Attribute {
+        key: "operation.name".to_string(),
+        value: AttributeValue::String("list_blobs".to_string()),
+    },
```

Comment:
I appreciate this pattern might exist with otel and maybe even tracing, but could we present this as a `HashMap<String, ???>` for more idiomatic use? I especially bring this up because we've been discussing it with Storage who wanted idiomatic tags but authored their service spec as an array of objects with "key" and "value" fields as opposed to, say, a `Record<T>` (to be fair, it's a very old spec).